worker-workflow/src/main/resources/workflow-control.js

```diff
@@ -261,6 +261,37 @@ function isLastAction(action) {
     return ACTIONS[ACTIONS.length - 1 ].name === action;
 }
 
+function onProcessDocument(e) {
+    processWorkersVersions(e.rootDocument);
+}
+
+function processWorkersVersions(document) {
+    var arrayOfWorkersVersions = [];
+    var currentSourceInfoWorkerName = getCurrentWorkerName(document);
+    var currentSourceInfoWorkerVersion = getCurrentWorkerVersion(document);
+
+    if (fieldExists(document, "PROCESSING_WORKER_VERSIONS")) {
+        var arrayOfWorkersVersions = getAllWorkerVersions(document.getField("PROCESSING_WORKER_VERSIONS").getValues());
+        var positionInArrayOfCurrentWorkerVersion = 
+                getPositionInArrayOfCurrentWorkerVersion(arrayOfWorkersVersions, currentSourceInfoWorkerName);
+        if (positionInArrayOfCurrentWorkerVersion === -1) {
+            var workerVersion = createWorkerVersionObject(currentSourceInfoWorkerName, currentSourceInfoWorkerVersion);
+            arrayOfWorkersVersions.push(workerVersion);
+        } else {
+            var workerVersionRetrieved = arrayOfWorkersVersions[positionInArrayOfCurrentWorkerVersion];
+            var currentVersion = currentSourceInfoWorkerName + " " + currentSourceInfoWorkerVersion;
+            if (workerVersionRetrieved.VERSION !== currentVersion) {
+                workerVersionRetrieved = createWorkerVersionObject(workerVersionRetrieved.NAME, currentSourceInfoWorkerVersion);
+            }
+            arrayOfWorkersVersions[positionInArrayOfCurrentWorkerVersion] = workerVersionRetrieved;
+        }
+    } else {
+        var workerVersion = createWorkerVersionObject(currentSourceInfoWorkerName, currentSourceInfoWorkerVersion);
+        arrayOfWorkersVersions = [workerVersion];
+    }
+    setFieldUsingAnArray(document, arrayOfWorkersVersions);
```

Comment:
It looks like this is always going to cause change log entries even when they are unnecessary.